### ISSUE TITLE
Transfer learning

### ISSUE THREAD

**Issue by Ciaran1981** [weecology/DeepForest#360]
**Describe the bug**

I'm not sure if this is a bug or my ignorance, but would really appreciate the help. Whenever I attempt to train the release model regardless of the no of epochs etc. it produces identical predictions to the model prior to any additional training. Can anyone explain this behavior / offer a solution?

Is there something further I need to do with the model object post training?

The training file and image are provided here (inside the zip is a shapefile of polygons). 

https://drive.google.com/drive/folders/1Qn8UWM_Zq0vUiSR0DyjtY28eS9MPTMqa?usp=sharing

I am aware they are not enough label-wise but I needed this to post a portable example. I have omitted the 'admin code' around the files and dirs, but can provide a full script. I have essentially followed the example. 

**To Reproduce**
```
ptch = 3000
ovlp = 0.05

m = main.deepforest()
m.use_release()
m.config["train"]["csv_file"] = annotations_file
m.config["train"]["root_dir"] = crop_dir
m.config["save-snapshot"] = False
m.config["validation"]["csv_file"] = validation_file
m.config["validation"]["root_dir"] = os.path.dirname(validation_file)

#create a pytorch lighting trainer used to training 
m.create_trainer()

start_time = time.time()
m.trainer.fit(m)
print(f"--- Training on GPU: {(time.time() - start_time):.2f} seconds ---")

df = m.predict_tile(inras, patch_size=ptch, patch_overlap=ovlp)
gdf = utilities.annotations_to_shapefile(df, transform=tranb, crs=br.crs)
```

**Environment**
 - OS: 
 ubuntu/linux
 - Python version and environment : 
 Python 3.9.9 packaged by conda-forge

**Comment by bw4sz** [comment]
hard to guess at what's happening here, but I can confirm that the code looks adequate. Can you do a dummy test and way overfit on your data? change

```
config["train"]["lr"] = 0.5
```
and do that for a bunch of epochs. This will massively overfit and should produce new (and terrible) predictions.

**Comment by Ciaran1981** [comment]
Hi - it seems by doing the following the transfer learning works ( I had previously not initialized the following - not sure if it is in the example code or not....)

`m.train()`

**Comment by bw4sz** [comment]
Happy to help here, let me know why this was re-opened.

**Comment by Ciaran1981** [comment]
Thanks - I though I'd solved it but alas I have not. Not got round to testing further yet.

**Comment by VinayakRaoDikshit** [comment]
Hello, @Ciaran1981 I would like to take this issue, can someone tell me that status whether further work is required or not. If yes, can you assign it to me ?

**Comment by Ciaran1981** [comment]
HI [VinayakRaoDikshit](https://github.com/VinayakRaoDikshit), yes by all means take it on. I don't know how to assign it to you though. The link to the data I used is still valid.

**Comment by ethanwhite** [comment]
I've assigned the issue to you @VinayakRaoDikshit - let us know if we can be of any help as you work on it.

**Comment by bw4sz** [comment]
We are moving issues like this to discussions. Happy to re-open and help there. Issues often get lost, they will get more traction as discussions.
